overhangio/tutor-minio
# feat: add separated grades bucket and manage querystring by config var

# Description

This PR pretends to add separated bucket for grades because the data store there can be sensitive.

Tutor-minio currently have the [default bucket public](https://github.com/overhangio/tutor-minio/blob/master/tutorminio/templates/minio/tasks/minio/init.sh#L6) and the other ones private, the idea of this PR is set `AWS_QUERYSTRING_AUTH` editable and `True` by default, because the querystring is needed to access to private buckets.

> If [custom_domain](https://github.com/jschneier/django-storages/blob/1.8/storages/backends/s3boto3.py#L604) is send, it will return the url before add the [querystring](https://github.com/jschneier/django-storages/blob/1.8/storages/backends/s3boto3.py#L615). But with this change we need to set `AWS_S3_SECURE_URLS` because s3 [enforces https method](https://github.com/jschneier/django-storages/blob/1.8/storages/backends/s3boto3.py#L255).

# How to test it

You can install this branch and with a basic tutor installation test the changes. Here is some screenshots of mi own test on local.

1. Grades bucket is private
![image](https://github.com/overhangio/tutor-minio/assets/39854568/080cf3cb-340c-479e-954c-3d85db607c37)

2. I can download the file without problem, in the url we can look the querystring 
![image](https://github.com/overhangio/tutor-minio/assets/39854568/27d7a65a-1042-45c0-9c78-4acc3b1495cf)

3. I can upload and look the profile image
![image](https://github.com/overhangio/tutor-minio/assets/39854568/c3ff0681-9429-44f4-bf7b-8e2c8c31b1aa)

4. For local purposes, the method is `http` and no querystring is sent as expected
![image](https://github.com/overhangio/tutor-minio/assets/39854568/a0c4daac-f04a-4b8e-bb34-2623120f4a78)
